ckanext/validation/logic.py

```diff
@@ -174,6 +175,43 @@ def resource_validation_show(context, data_dict):
 
     return _validation_dictize(validation)
 
+def resource_table_schema_infer(context, data_dict):
+    '''
+    Use frictionless framework to infer a resource schema
+    '''
+
+    t.check_access('resource_create', context, data_dict)
+
+    if not data_dict.get(u'resource_id'):
+        raise t.ValidationError({u'resource_id': u'Missing value'})
+
+    store_schema = data_dict.get('store_schema', True)
+
+    resource = t.get_action('resource_show')(
+        {}, {u'id': data_dict['resource_id']})
+
+    source = None
+    if resource.get('url_type') == 'upload':
+        upload = uploader.get_resource_uploader(resource)
+        if isinstance(upload, uploader.ResourceUpload):
+            source = upload.get_path(resource['id'])
+
+    if not source:
+        source = resource['url']
+
+    with system.use_context(trusted=True):
+        # TODO: check for valid formats
+        fric_resource = Resource({'path': source, 'format': resource.get('format', 'csv').lower()})
+        fric_resource.infer()
+        resource['schema'] = fric_resource.schema.to_json()
```

Comment:
Can we get an error here?

Reply:
Ok so let's catch these exceptions and wrap them in a raised `ValidationError`

Reply:
I did that on the other PR https://github.com/frictionlessdata/ckanext-validation/pull/83/files#diff-a6aa6aa6d2775f6c7180f2aa7d8e6a6220c22402b6094503669b990e6edefd48R220